lib/iris/tests/unit/mesh/components/test_MeshCoord.py

```diff
@@ -751,20 +775,22 @@ def fetch_sources_from_mesh():
         # Check all the source coords are lazy.
         for coord in fetch_sources_from_mesh():
             # Note: not all are actual Coords, so can't use 'has_lazy_points'.
-            self.assertTrue(is_lazy_data(coord._core_values()))
+            assert is_lazy_data(coord._core_values()) is True
 
         # Calculate both points + bounds of the meshcoord
-        self.assertTrue(meshcoord.has_lazy_points())
-        self.assertTrue(meshcoord.has_lazy_bounds())
-        meshcoord.points
-        meshcoord.bounds
-        self.assertFalse(meshcoord.has_lazy_points())
-        self.assertFalse(meshcoord.has_lazy_bounds())
+        assert meshcoord.has_lazy_points() is True
+        assert meshcoord.has_lazy_bounds() is True
+        mc_points = meshcoord.points
+        mc_bounds = meshcoord.bounds
+        assert meshcoord.has_lazy_points() is True
+        assert meshcoord.has_lazy_bounds() is True
```

Comment:
Is it expected that these remain lazy? If so this is definitely worth mentioning in the whatsnew (which is worth adding)

Reply:
Yeah, that's a good point. 

In discussion with @pp-mo, we thought the best contract was that it returns real data, but the points remain lazy. 

Otherwise, you're realizing the coord_on_mesh points as well, which you shouldn't be able to do accidentally. 

The whatsnew is gonna be a behemoth I fear, on it now. Ill try and pare it down